### ISSUE TITLE
Edited XML does not saving

### ISSUE THREAD

**Issue by aafanasyev** [retspen/webvirtmgr#489]
Hi everyone,

I am facing strange problem with XML here.

Due some network limitation in my  KVM setup I need to use a NAT based on the virbr0.
Therefore, I would like to forward a some ports from my guests to other world.
I found this nice article:
http://secomputing.co.uk/2012/02/21/Forwarding-ports-to-KVM-clients/

I did all required changes according this article:
```
<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'> 

<qemu:commandline> 
   <qemu:arg value='-redir'/>
   <qemu:arg value='tcp:22::22'/>
</qemu:commandline>`
```

However, when I click save button in Instance>Settings>XML the changes does not save if I recheck it later.

Why? How can I solve it?

Thank you.

P.S.: I have Ubuntu 14.04.1 LTS. During installation the webvirtmgr was hosted in /var/www and not in /var/www/webvirtmgr/ All paths were changed respectively and I did not faced any issues before this one.

**Comment by sammcj** [comment]
+1 I too am experiencing this.
It looks like even if you edit the xml by hand it still doesn't update?

**Comment by aafanasyev** [comment]
I do not have source by my hand, but I red somewhere that if syntax not right or depreciated the xml will be not updated. Moreover it can lead to distraction of the instance and virtual hard disc (I experienced it by myself). So it is not an issue I think it is more version difference. And I solved it finally like this:
http://ubuntuforums.org/showthread.php?t=2261173&p=13210545

**Comment by sammcj** [comment]
That's correct, however it would be good behaviour to pop a warning if
the XML is somehow malformed or something like that?

-- 
Sam McLeod
@s_mcleod[https://twitter.com/s_mcleod]

**Comment by aafanasyev** [comment]
Yes, definitely! However, I think it will nice if some kind of spellchecker or code highlighter will be integrated too, but it is not easy.

**Comment by sammcj** [comment]
Syntax highlighting should be easy enough.

It's a bloody nice project actually - I only just tried it today.

-- 
Sam McLeod
@s_mcleod[https://twitter.com/s_mcleod]
